mindsdb/api/executor/planner/plan_join.py

```diff
@@ -279,49 +278,24 @@ def _check_node_condition(node, **kwargs):
         self.query_context["binary_ops"] = binary_ops
 
     def check_use_limit(self, query_in, join_sequence):
-        """
-        Determine if LIMIT can be pushed down to the first table fetch.
-
-        LIMIT pushdown means: fetch only N rows from the first table, then join.
-        This optimization is ONLY correct when the first table determines the final row count.
-
-        LIMIT pushdown is CORRECT for:
-        - Single table queries (no join)
-        - LEFT JOIN (left table determines row count - each left row appears exactly once)
-        - Joins with ML predictors
-
-        LIMIT pushdown is SLOW for:
-        - INNER JOIN between tables
-        - RIGHT JOIN (right table determines row count, not left)
-
-        When LIMIT pushdown is disabled, we fetch all data and apply LIMIT after the join.
-        This is slower but guarantees correct results.
-        """
+        # if only models (predictors), not for regular table joins
         use_limit = False
         if query_in.having is None and query_in.group_by is None and query_in.limit is not None:
             use_limit = True
-
-            # Check what we're joining
-            has_predictor = False
-            cannot_pushdown_limit = False
+            has_join = False
+            regular_table_count = 0
 
             for item in join_sequence:
                 if isinstance(item, TableInfo):
-                    if item.predictor_info is not None:
-                        has_predictor = True
+                    # Check if it's a regular table (not a predictor, not a subselect)
+                    if item.predictor_info is None and item.sub_select is None:
+                        regular_table_count += 1
                 elif isinstance(item, Join):
-                    join_type = (
-                        item.join_type.upper() if hasattr(item.join_type, "upper") else str(item.join_type).upper()
-                    )
-
-                    # LEFT JOIN preserves left table row count - LIMIT pushdown is safe
-                    if join_type in ("LEFT JOIN", "LEFT OUTER JOIN"):
-                        continue
+                    has_join = True
```

Comment:
`if join_type in ("LEFT JOIN", "LEFT OUTER JOIN"):` - this condition was removed
is it braking something?

Reply:
Its not breaking since it is safe due to LEFT joins is safe for LIMIT but may be slower. I think we can add back this optimization